openfl-tutorials/interactive_api/PyTorch_MVTec_PatchSVDD/workspace/PatchSVDD_with_Director.ipynb

```diff
@@ -0,0 +1,1211 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "liquid-jacket",
+   "metadata": {},
+   "source": [
+    "# Federated PatchSVDD algorithm with Director example\n",
+    "## Using low-level Python API"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "83e6aae2",
+   "metadata": {},
+   "source": [
+    "# PatchSVDD algorithm\n",
+    "Anomaly detection involves making a binary decision as to whether an input image contains an anomaly, and anomaly segmentation aims to locate the anomaly on the pixel level. The deep learning variant of Support vector data description (SVDD: a long-standing algorithm used for anomaly detection) is used to the patch-based method using self-supervised learning. This extension enables anomaly segmentation and improves detection performances which are measured in AUROC on MVTec AD dataset.\n",
+    "\n",
+    "* Original paper: https://arxiv.org/abs/2006.16067\n",
+    "* Original Github code: https://github.com/nuclearboy95/Anomaly-Detection-PatchSVDD-PyTorch/tree/934d6238e5e0ad511e2a0e7fc4f4899010e7d892\n",
+    "* MVTec ad dataset download link: https://www.mydrive.ch/shares/38536/3830184030e49fe74747669442f0f282/download/420938113-1629952094/mvtec_anomaly_detection.tar.xz"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "af0579f8",
+   "metadata": {},
+   "source": [
+    "# Long-Living entities update\n",
+    "\n",
+    "* We now may have director running on another machine.\n",
+    "* We use Federation API to communicate with Director.\n",
+    "* Federation object should hold a Director's client (for user service)\n",
+    "* Keeping in mind that several API instances may be connected to one Director.\n",
+    "\n",
+    "\n",
+    "* We do not think for now how we start a Director.\n",
+    "* But it knows the data shape and target shape for the DataScience problem in the Federation.\n",
+    "* Director holds the list of connected envoys, we do not need to specify it anymore.\n",
+    "* Director and Envoys are responsible for encrypting connections, we do not need to worry about certs.\n",
+    "\n",
+    "\n",
+    "* Yet we MUST have a cert to communicate to the Director.\n",
+    "* We MUST know the FQDN of a Director.\n",
+    "* Director communicates data and target shape to the Federation interface object.\n",
+    "\n",
+    "\n",
+    "* Experiment API may use this info to construct a dummy dataset and a `shard descriptor` stub."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "alike-sharing",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Install dependencies if not already installed\n",
+    "!pip install torchvision==0.8.1\n",
```

Comment:
Can we install all requirements in a single line? 
Like: `pip install torchvision==0.8.1 torch tqdm`